govcd/api_vcd.go

```diff
@@ -290,6 +250,23 @@ func NewVCDClient(vcdEndpoint url.URL, insecure bool, options ...VCDClientOption
 			panic(fmt.Sprintf("unable to initialize vCD client: %s", err))
 		}
 	}
+
+	// optionally load certs.  Might be better to do this upfront rather than replace the transport, but do
+	// not with to change the order at which the options are read in
+	disableCompression := vcdClient.Client.OauthUrl != ""
+	if vcdClient.Client.ClientTlsCert != "" {
+		util.Logger.Printf("[OAUTH] setting up TLS certs")
+		x509cert, err := tls.X509KeyPair([]byte(vcdClient.Client.ClientTlsCert), []byte(vcdClient.Client.ClientTlsKey))
+		if err != nil {
+			panic(fmt.Errorf("Unable to load key pair: %v", err))
+		}
+		certs := []tls.Certificate{x509cert}
+		vcdClient.Client.Http.Transport = &http.Transport{
+			TLSClientConfig:    &tls.Config{Certificates: certs, InsecureSkipVerify: insecure},
+			DisableCompression: disableCompression,
```

Comment:
You've lost the Proxy and TLSHandhakeTimeout settings set earlier (even though they're probably the defaults so it doesn't matter). But it might be better to preserve the existing Transport obj, and just do:
```
vcdClient.Client.Http.Transport.TLSClientConfig = &tls.Config{Certificates: certs, InsecureSkipVerify: insecure}
vcdClient.Client.Http.Transport.DisableCompression = disableCompression
```